### DISCUSSION TITLE
questions about p mechanism

### DISCUSSION THREAD

**Discussion by zhaob1n** [helix-editor/helix#3402] [Q&A]
Say, I copy a line by `xy`
When I paste it, instead of pasting directly after the cursor, it opens a  new line and pastes  to the line after the cursor.
I think it's quite interesting and actually saves effort. But I don't get how it works.

**Comment by axdank** [comment]
if I am not mistaken, I think that when the copied line has the line break `\n` is the case that copies up or down in a new line, in the case that the copied does not have the \n at the end, paste without problems before or after in the position in which your cursor is.
